### ISSUE TITLE
Native OTR support?

### ISSUE THREAD

**Issue by Mikaela** [weechat/weechat#298]
Having native OTR support would increase privacy (and properly verified) ensure that you are talking with who you think in the discussions and other clients are planning to do this by default including irssi (I think the code already exists) and Pidgin.

I know that OTR support is available with `/script install otr.py`, but it's a script, not plugin and doesn't come by default and it's not as popular as scripts like `buffers.pl` and `iset.pl` which almost everyone probably have.

**Comment by tribut** [comment]
May someone (Mikaela?) could write an entry to the FAQ, making it easier for users to discover that otr.py exists?

**Comment by Mikaela** [comment]
I think I should be able to do that. Is it just editing the faq.rst or should I also do something to the different language versions?

**Comment by Mikaela** [comment]
Which category should it go to? https://weechat.org/files/doc/weechat_faq.en.html

IRC or scripts?

**Comment by Shawn-Smith** [comment]
I don't see the need for this, since it's provided in a script. You don't see people asking for `buffers.pl` to be rewritten as a plugin and it's one of the most popular scripts out there.

I don't see it getting used very much even if it is made a plugin.

**Comment by Mikaela** [comment]
> I don't see the need for this, since it's provided in a script.

The script is not so widely used and probably won't become so widely used. Most of people are using every plugin that come with WeeChat.

It was also extension for irssi and is plugin for Pidgin, but the first one already has OTR support in git if I understood correctly and the second has it planned.

> You don't see people asking for buffers.pl to be rewritten as a plugin and it's one of the most popular scripts out there.

Actually I do...

```
2014-10-03.log:[00:06:49] <gb> nils written the buffers plugin yet?
2014-12-04.log:[00:19:09] <nils_2> i would prefer a buffers plugin :p 
2015-01-05.log:[20:29:00] <sq> I hope 1.3 will feature buffers.pl as a plugin :s
2015-01-06.log:[21:57:31] <gb> it'll be nice when it and buffers.pl are both plugins that ship by default :)
```

Buffers.pl is the script that is asked to be made a plugin the most and I think @FlashCode has hinted about it happening in the future. You can probably find more instances in your logs, that is so much requested.

> I don't see it getting used very much even if it is made a plugin.

Probably depends. 

* Would it advertise the OTR capability by default?
    * Yes: every WeeChat user who PMs each other automatically triggers OTR.
    * No: Would it be option, but default to false?
        * Yes: Multiple users would change it to true and as they PM other people using WeeChat, they notice the whitespace tag and activate OTR.

**Comment by Mikaela** [comment]
* The irssi issue: https://github.com/irssi/irssi/issues/196

**Comment by JohnMaguire** [comment]
:+1:

**Comment by JohnMaguire** [comment]
I agree with this. I understand that it's provided as a script, but let's face a couple facts:

1. Most users aren't familiar with encryption, how it works, or why and when it is useful.
2. Those that are also generally know that encryption can be difficult, painful, and if done incorrectly, useless, to setup.

By packaging OTR with the clients, it's a signal that encryption is both important, and should be on-by-default. More and more services are moving to an idea of encryption by default, and I think any communication channel that *can* support encryption, *should* support encryption *by default*.

**Comment by JohnMaguire** [comment]
To clarify, I think it being enabled by default is important because encryption requires that both parties participate. Because OTR adoption is relatively low, it's currently not very useful. By increasing adoption, OTR becomes far more useful. *I* want to use encryption. I can't because most don't care.

**Comment by wnereiz** [comment]
I plan to work on it (but can not promise) - to support otr natively - with libotr.
One reason is that otr.py based on pure-python-otr project which lack of maintenance and has been considered as [insecure by the developer](https://github.com/python-otr/pure-python-otr#this-software-is-experimental-and-potentially-insecure-do-not-rely-on-it)

**Comment by weechatter** [comment]
it exists an old otr-plugin for weechat (from 2009). i have the source code, but can not remember where i get it from

**Comment by wnereiz** [comment]
Start working on it. But I found the old otr-plugin project is out of the date. The api for libotr changed much, a lot of places need to be change. And otr-plugin are not only for weechat, so there are some dynamic generated code. So it will take quite a long time for me on the task.

**Comment by polyzen** [comment]
`otr.py` was updated within the past two months. OTR seems more and more niche by the day; the script then should be sufficient. Should this be closed?

Also, Python 3 is upon us:
https://pythonclock.org/
https://github.com/python-otr/pure-python-otr/issues/58

**Comment by RaitoBezarius** [comment]
I don't believe it's niche. It's 2020 and I'm very still interested into native OTR.

**Comment by ernestask** [comment]
Niche or not, there is essentially no good known solution. otr.py just reeks of old and obsolete by now. Having something native upstream would at least focus the efforts.

**Comment by kpcyrd** [comment]
I agree, I wouldn't recommend otr.py since the crypto library it's using seems to be dead and might have security issues:

https://github.com/python-otr/pure-python-otr/issues/75
https://github.com/python-otr/pure-python-otr/issues/68

**Comment by weechatter** [comment]
plugin or script!? ;-)

**Comment by ernestask** [comment]
The idea is to use libotr directly, so a script doesn’t make sense.
